### ISSUE TITLE
`Byte_Representation` gadget may lead to vk change against each assignment

### ISSUE THREAD

**Issue by noel2004** [scroll-tech/mpt-circuit#31]
In `Byte_Representation` gadget, the `selector` (a fixed col) is enabled while assigning byte represents:
https://github.com/scroll-tech/mpt-circuit/blob/2cc63d2b765a20cd12136d1df675052c9353a582/src/gadgets/byte_representation.rs#L107

It should be supposed that we have diffferent counts on assigned bytes for synctheis base on the witness. That is, the circuit (vk and pk) would change with changing on witness and the test `vk_validity` would expect to fail

**Comment by z2trillion** [comment]
Yes, this is problem now. I plan to solve this by padding this gadget (and all other gadgets) with default assignments to a fixed length.

For this gadget, assigning 0 to every advice column should satisfy the constraints.
